File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        res = float("inf")

        for i in range(n):
            sumsum = 0 
            for j in range(i,n):
                sumsum += nums[j]
                if target <= sumsum:
                    res = min(res,j-i+1)
                    break
        return 0 if res == float("inf") else res
                     

File: Data Structures & Algorithms/binary-search/submission-2.cs
public class Solution {
    public int Search(int[] nums, int target) 
    {
        int leftbound = 0;
        int rightbound = nums.Length-1; 
        
        while(leftbound <= rightbound)
        {
            int m = leftbound +((rightbound - leftbound) / 2) ;
            if(nums[m] < target) leftbound = m+1;
            else if (nums[m] > target) rightbound = m-1;
            else return m;

        }
        return -1; 

    }
}
